packages/components/docs/localDev.md

```diff
@@ -0,0 +1,261 @@
+some guides to best practices for doing front-end development:
+* [Alan's Guidelines](https://docs.google.com/presentation/d/1hW9gYbWhW6spr7uhAjpKBNl3hWPXUOPbmlwD6UkYWE8/edit?pli=1#slide=id.g5627a1f538_0_120)
+* [Airbnb Style Guide](https://github.com/airbnb/javascript)
+* [Immer for immutability](./immer.md)
+* [QueryModel API](./QueryModel.md)
+* [Jest Testing Recommendations](./jest.md)
+
+Generally, when doing development, you should:
+
+* Place source code in a `src` subdirectory (note the distinction between the `internal` code and `public` code.
+* Place typing files in a `src/typings` subdirectory.
+* Add documentation a-plenty and update the package [release notes](../releaseNotes/labkey/components.md) as you add/fix/update the package.
+
+### Local Development
+
+When making modifications to the existing package, you should:
+* Use an alpha version number for the package of the form `X.Y.Z-fb-my-branch-name.0`, where `X.Y.Z` is your best guess at the
+next [SemVer](https://semver.org/) version that will include your changes, `fb-my-branch-name` is the name of your
+feature branch with underscores replaced by hyphens and the `.0` is just a starting point for the prerelease versioning. This `.0`
+version will be incremented with each alpha package version you want to publish and test in your LabKey module.
+Note that the version number within the `package.json` file will be set while running the  `yarn publish` command.
+See [below](#version-numbering) for more on version numbering.
+* Update the `releaseNotes/labkey/components.md` file to document what is changing in this version. Note that the final release
+version number and date will be set just before you merge your feature branch.
+* Write [jest](https://jestjs.io/docs/en/getting-started.html) tests for your React components, models, and utility functions.
+    * See additional documentation on [Jest testing recommendations](./jest.md).
+* Write or update [storybook stories](#storybook) that illustrate the functionality.  This is the easiest way to do the bulk of manual
+testing and iteration on display updates.
+    * We have several examples of stories that use actual data captured
+from the server for various Ajax calls that are required.
+* Test within the application, using a published alpha package version, once display and functionality are as expected from within storybook.
+
+#### Getting @labkey/components packages to the application
+
+While you can [publish](#publishing) a pre-release version of the components packages and then update your application's `package.json`
+to reference the new pre-release version in order to view the changes within the application, you will likely want to
+be able to do this without publishing for quicker development iteration.
+
+In order to use and test the components you are developing or modifying in this repository within another application,
```

Comment:
Oh, we should update this to recommend the linking stuff.

Reply:
done